src/ballet/txn/fd_txn.h

```diff
@@ -57,12 +57,15 @@
    valid transaction can have is 12 signatures. The most I've seen in practice
    is about 7.
 
+   FD_TXN_ACTUAL_SIG_MAX: used to allocate arrays of signatures, pubkeys, etc.
+
    From the spec: "The Solana runtime verifies that the number of signatures
    [stored as a compact-u16] matches the number in the first 8 bits of the
    message header."
    Thus this value must live in the range where compact-u16 and uint8
    representations are identical, hence a max of 127. */
 #define FD_TXN_SIG_MAX               (127UL)
+#define FD_TXN_ACTUAL_SIG_MAX        (12UL)
```

Comment:
Where does 12UL come from?

Reply:
It's in the comment - given the max tx size, you can only fit 12x 64-bit signatures.
I personally think it's better to keep the constants close, so we don't risk to change one but not the other.

Reply:
Ah, didn't see that thanks